code/game/objects/items/weapons/shields.dm

```diff
@@ -15,70 +64,70 @@
 	w_class = 4
 	materials = list(MAT_GLASS=7500, MAT_METAL=1000)
 	origin_tech = "materials=2"
-	attack_verb = list("shoved", "bashed")
-	var/cooldown = 0 //shield bash cooldown. based on world.time
+	slowdown_deployed = 2
+	block_chance_deployed = list(melee = 70, bullet = 60, laser = 40, energy = 30)
+	var/cooldown = 0
 
-/obj/item/weapon/shield/riot/attackby(obj/item/weapon/W, mob/user, params)
-	if(istype(W, /obj/item/weapon/melee/baton))
+/obj/item/weapon/shield/deployable/riot/attack_self(mob/living/user)
+	..()
+	block_push = active
+
+/obj/item/weapon/shield/deployable/riot/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/melee/baton) && active)
 		if(cooldown < world.time - 25)
 			user.visible_message("<span class='warning'>[user] bashes [src] with [W]!</span>")
 			playsound(user.loc, 'sound/effects/shieldbash.ogg', 50, 1)
 			cooldown = world.time
 	else
 		..()
 
-/obj/item/weapon/shield/riot/roman
-	name = "roman shield"
-	desc = "Bears an inscription on the inside: <i>\"Romanes venio domus\"</i>."
-	icon_state = "roman_shield"
-	item_state = "roman_shield"
+/obj/item/weapon/shield/deployable/riot/dropped(mob/user)
+	if(active && user)
+		attack_self(user)
 
-/obj/item/weapon/shield/energy
+/obj/item/weapon/shield/deployable/energy
 	name = "energy combat shield"
 	desc = "A shield capable of stopping most melee attacks. Protects user from almost all energy projectiles. It can be retracted, expanded, and stored anywhere."
-	icon = 'icons/obj/weapons.dmi'
-	icon_state = "eshield0" // eshield1 for expanded
+	icon_state = "eshield"
 	force = 3
 	throwforce = 3
 	throw_speed = 3
 	throw_range = 5
 	w_class = 1
 	origin_tech = "materials=4;magnets=3;syndicate=4"
-	attack_verb = list("shoved", "bashed")
-	var/active = 0
-
-/obj/item/weapon/shield/energy/hit_reaction(mob/living/carbon/human/owner, attack_text, final_block_chance)
-	if(active)
-		return ..()
-	return 0
+	block_chance = list(melee = 60, bullet = 70, laser = 60, energy = 40) //Energy var is irrelevant as far as I'm concerned due to IsReflect() (correct me if I'm wrong - might be good against disablers?)
+	force_deployed = 10
 
-/obj/item/weapon/shield/energy/IsReflect()
+/obj/item/weapon/shield/deployable/energy/IsReflect()
 	return (active)
 
-/obj/item/weapon/shield/energy/attack_self(mob/living/carbon/human/user)
-	if(user.disabilities & CLUMSY && prob(50))
-		user << "<span class='warning'>You beat yourself in the head with [src].</span>"
-		user.take_organ_damage(5)
+/obj/item/weapon/shield/deployable/energy/attack_self(mob/user)
+	var/mob/living/carbon/human/H = user
+	if(istype(H) && H.disabilities & CLUMSY && prob(50))
+		H << "<span class='warning'>You beat yourself in the head with [src].</span>"
+		H.take_organ_damage(force)
 	active = !active
 	icon_state = "eshield[active]"
 
 	if(active)
-		force = 10
+		block_chance = block_chance_deployed
+		slowdown = slowdown_deployed
+		force = force_deployed
```

Comment:
How come force has a force_deployed variable but throwforce, throw_speed, and w_class don't?